### ISSUE TITLE
Adapt for ember-data 1.0.0-beta16

### ISSUE THREAD

**Issue by rsutphin** [nolanlawson/ember-pouch#48]
Starting in ember-data 1.0.0-beta16, adapters will no longer receive actual records. They will receive snapshots instead. See emberjs/data#2883.

This is a placeholder for any work that's necessary to adapt to this — it's possible that no changes will be necessary.

**Comment by rsutphin** [comment]
It looks like the necessary changes will be [pretty extensive](http://emberjs.com/blog/2015/03/23/ember-data-1-0-beta-16-released.html) to avoid deprecation warnings. I'd vote to drop backwards compatibility and go to version 2.0.0.

**Comment by broerse** [comment]
`DEPRECATION: You called _createSnapshot on what's already a DS.Snapshot. You shouldn't manually create snapshots in your adapter since the store passes snapshots to adapters by default.` does not mean we have to go for a major version upgrade for every ember-data beta step but dropping backwards compatibility for snapshots seems OK to me. I don't know if it is the snapshots but ember-pouch does not seem aware of change you make in an other browser instance or new data coming in. See [http://exmer.com/bloggrcouch/](http://exmer.com/bloggrcouch/)  I will update to beta16 today to check if this helps. Ember data 1.0 will be released June 12 so  if we need a new major version perhaps on June 12 ;-)

**Comment by broerse** [comment]
@rsutphin ember-data beta16 does not solve my problem and switching back to 1.2.3 does not help so I think my problem is no a snapshot problem and does not belong in your issue. I have created a new issue for this: https://github.com/nolanlawson/ember-pouch/issues/50

**Comment by rsutphin** [comment]
> `DEPRECATION: You called _createSnapshot on what's already a DS.Snapshot. You shouldn't manually create snapshots in your adapter since the store passes snapshots to adapters by default.` does not mean we have to go for a major version upgrade for every ember-data beta step but dropping backwards compatibility for snapshots seems OK to me.

If there was just that one deprecation warning, then I wouldn't worry about it. But it seems from the [blog post](http://emberjs.com/blog/2015/03/23/ember-data-1-0-beta-16-released.html) that the API for a snapshot is entirely different from a model (`id` instead of `get('id')`, `attr` for scalar attributes, `hasMany` and `belongsTo` for associations, etc.) and the old versions will _all_ print deprecation warnings. If that's the case, then your logs will be filled with deprec warnings to the exclusion of everything else.

Anyway, just my vote for 2.0.0 if it does turn out to be a deprecation nightmare. Unfortunately I probably won't be able to help with this right away (unless it turns out I to need to update past beta.15 for another reason).

**Comment by broerse** [comment]
It seems Ember-Date 16.1 removed the Deprecation warnings?

**Comment by OleRoel** [comment]
As an intermediate solution before moving to a version 2.0.0 we could also extend the check for snapshot in [_recordToData](https://github.com/nolanlawson/ember-pouch/blob/master/dist/amd/pouchdb-adapter.js#L118):

```javascript
    if (typeof record.record === 'undefined' && typeof record._createSnapshot === 'function') {
      recordToStore = record._createSnapshot();
    }
```
The 'if' statement will make the deprecation message go away in 1.0.0-beta16 and keep the adapter compatible with 14 and 15.

This solution is just pragmatic and not beautiful and should be removed when EmberData is final.

(@ broerse: I still see the deprecation warnings in 1.0.0-beta16.1)

**Comment by broerse** [comment]
@OleRoel I was too fast. My deprecation warnings changed from always to sometimes but perhaps that was because I also update my application. @nolanlawson can you take a look or are you planning updates after 12 June ?

**Comment by rsutphin** [comment]
@OleRoel, I like that idea. Does it really get rid of all deprecation warnings? I was concerned more with the accessor-based ones, but maybe ember-pouch delegates all that to the serializer and so would be fine. (Unfortunately, I still haven't had a chance to try out beta.16 myself.)

**Comment by nolanlawson** [comment]
Sorry, I've been busy with other stuff these days. If y'all have time, a pull request would be awesome. :)

**Comment by OleRoel** [comment]
I'll take the PR.

**Comment by broerse** [comment]
@OleRoel Thanks. I am a beginner but can test it for you!

**Comment by OleRoel** [comment]
@broerse thank you for your offer! I am going to test my code tomorrow.

**Comment by broerse** [comment]
@OleRoel Your fix is working fine! Just updated to 1.2.5

**Comment by nolanlawson** [comment]
Cool beans. :)
